### DISCUSSION TITLE
Extracting value from API response

### DISCUSSION THREAD

**Discussion by rpmcoding** [Budibase/budibase#7918] [Help] [answered]
I am having trouble extracting the value of sign_page_url from an API response so that I can then use it in the app to redirect the user to that url.

This is the response json.

`{
   "status":"queued",
   "data":{
      "contract":{
         "id":"3ca4cefe-1d2b-460a-897c-82cdc6c87f94",
         "status":"sent",
         "title":"Loan Agreement - Saver package",
         "metadata":"ID0001",
         "source":"api",
         "test":"yes",
         "company_name":"WhiteLabel LLC",
         "custom_webhook_url":"https://your-website.com/esignaturesio-custom-webhook",
         "placeholder_fields":[
            {
               "api_key":"interest_rate",
               "value":"3.2%"
            },
            {
               "api_key":"floor-plan"
            }
         ],
         "signers":[
            {
               "id":"df2ccc38-410f-4d1a-9ff2-f944927d29bf",
               "name":"Benny Bell",
               "email":"gfdsg@fdsaf.com",
               "mobile":"+44545454540",
               "company_name":"ACME Corp",
               "signing_order":"1",
               "auto_sign":"no",
               "redirect_url":"https://your-website.com/aftersign",
               "required_identification_methods":[
                  "email",
                  "sms"
               ],
               "signature_request_delivery_method":"embedded",
               "signed_document_delivery_method":"email",
               "sign_page_url":"https://esignatures.io/sign/df2ccc38-410f-4d1a-9ff2-f944927d29bf"
            }
         ]
      }
   }
}`

Please can someone assist?

**Comment by melohagan** [comment, marked as answer]
Hey @rpmcoding 

This should do the trick in the transformer: `data.data.signers[0].sign_page_url`

**Reply by rpmcoding** [reply]
@melohagan thanks.  When I add this to the data transformed, it does work with return `data.data.contract.signers[0].sign_page_url` .  However, the json is returned by the transformer: 

{"value":"https://esignatures.io/sign/07bbx07x21-2b9xa-4d56-x8a2b-c37be1x81f1c"}

How do I get it so it just returns https://esignatures.io/sign/07bbx07x21-2b9xa-4d56-x8a2b-c37be1x81f1c ?

**Reply by melohagan** [reply]
I think the API will always return an object because it needs fields so that it can be represented in a table.
E.g. 

| Value |
| --- |
| url |

If you want to use it in the **Navigate To** action then it'll be something like the following *On Click* event:
 - **Execute Query** (That returns the sign_page_url)
 - **Navigate To** `{{ Action 1.Query result.Value }}`


![Screenshot 2022-09-27 at 14 23 59](https://user-images.githubusercontent.com/101575380/192538433-5d9aaaa8-d113-4761-9a65-ec58c9d1019b.png)

**Reply by rpmcoding** [reply]
Thanks that led me down the right path, thanks again.

For anyone reading this later on with a similar problem the code which worked was:  {{ Action 1.Query result.data[0].value}}
